cmd/zc_traverser_list.go

```diff
@@ -73,15 +73,12 @@ func (l *listTraverser) traverse(processor objectProcessor, filters []objectFilt
 		// case 2: child2 is a directory, and it has items under it such as child2/grandchild1
 		//         the relative path returned by the child traverser would be "grandchild1"
 		//         it should be "child2/grandchild1" instead
-		preProcessor := func(object storedObject) error {
+		childPreProcessor := func(object *storedObject) {
 			object.relativePath = common.GenerateFullPath(childPath, object.relativePath)
-			return processor(object)
 		}
+		preProcessorForThisChild := preprocessor.FollowedBy(childPreProcessor) // always chain pre-processors, never replace them (this is so we avoid making any assumptions about whether an old processor actually exists)
```

Comment:
>// always chain pre-processors, never replace them (this is so we avoid making any assumptions about whether an old processor actually exists) [](start = 73, length = 142)

Consider moving this comment to .FollowedBy (or adding it wherever .FollowedBy is used)

Reply:
Good idea. Have moved it.